sysvabi64/sysvabi64.rst

```diff
@@ -916,10 +923,639 @@ These follow the AAELF64_ base definition from Arm.
+the address of a function both the shared library and the application
+must use the same address for the functions for address comparisons to
+work as expected. If the static linker uses a PLT entry for a function
+as the address for a function then it must place the address of the
+PLT for the function in the dynamic symbols table entry for the
+function. This will result in a dynamic symbol table entry with a
+section index of ``SHN_UNDEF``, a type ``STT_FUNC`` and a non-zero
+``st_value``. A reference to this symbol will be resolved by the
+dynamic linker to the address PLT for the function in the executable.
+
+The ``R_<CLS>_JUMP_SLOT`` relocations defined in AAELF64_ are
+associated with PLT entries. These entries are used for direct
+function calls rather than for references to function addresses. These
+relocations do not use the special symbol value described
+above. Otherwise a very tight endless loop would result.
+
+In a position independent executable all function addresses have
+dynamic relocations so no indirection via the PLT entry is necessary.
+
+Shared objects may implement direct function calls to non-pre-emptable
+symbols using the BL instruction. A symbol is not pre-emptable if:
+
+ * It has ``STB_LOCAL`` binding.
+
+ * It has a symbol visibility that is not ``STV_DEFAULT``.
+
+ * It has been given the ``local`` symbol version, defined in
+   SYM-VER_.
+
+All function calls to pre-emptable symbols must be indirected through
+the static linker created Procedure Linkage Table.
+
+Procedure Linkage Table
+-----------------------
+
+Much as the global offset table (GOT) redirects position independent
+address calculations to absolute locations, the Procedure Linkage
+Table (PLT) redirects position independent function calls to absolute
+locations. The static linker cannot resolve execution transfers (such
+as function calls) from one executable or shared object to
+another. Consequently, the static linker arranges to have the program
+transfer control to entries in the PLT. In the AArch64 architecture,
+PLTs reside in the shared text segment, but they in turn use addresses
+in the per-process global offset table section ``.got.plt``. The
+dynamic linker determines the destinations' absolute addresses and
+modifies the global offset table entry accordingly. The dynamic linker
+can thus redirect the entries without compromising the position
+independence and shareability of the program's text. Executable files
+and shared objects have their own separate procedure linkage
+tables. The same procedure linkage table format is used for both
+executables and shared objects.
+
+Following the steps below, the dynamic linker and the program
+cooperate to resolve symbolic references to functions through the
+procedure linkage table and the global offset table.
+
+ * When first creating the memory image of the program, the dynamic
+   linker sets the second and the third entries in the global offset
```

Comment:
Here the "global offset table" refers to `.got.plt`.

Reply:
Ack, will clarify.